dex/src/main/java/io/crate/data/join/JoinBatchIterator.java

```diff
@@ -113,10 +113,7 @@ public void kill(@Nonnull Throwable throwable) {
     }
 
     @Override
-    public boolean involvesIO() {
-        if (left.involvesIO()) {
-            return true;
-        }
-        return right.involvesIO();
+    public boolean hasLazyResultSet() {
+        return left.hasLazyResultSet() || right.hasLazyResultSet();
```

Comment:
Shouldn't it return `false` if one is `false`?

Reply:
I think it's safer to weight the true / lazy case higher. If you insert from a join of `values` with `generate_series` it should throttle.